athena-mysql/src/main/java/com/amazonaws/athena/connectors/mysql/MySqlMetadataHandler.java

```diff
@@ -215,11 +218,47 @@ public GetSplitsResponse doGetSplits(
         return new GetSplitsResponse(getSplitsRequest.getCatalogName(), splits, null);
     }
 
+    private List<TableName> getPaginatedTables(Connection connection, String databaseName, int token, int limit) throws SQLException
+    {
+        String tablesAndViews = "Tables and Views";
+        String sql = "SELECT table_name as \"TABLE_NAME\", table_schema as \"TABLE_SCHEM\" FROM information_schema.tables WHERE table_schema = ? ORDER BY TABLE_NAME LIMIT ?, ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setString(1, databaseName);
+        preparedStatement.setInt(2, token);
+        preparedStatement.setInt(3, limit);
+        LOGGER.debug("Prepared Statement for getting tables in schema {} : {}", databaseName, preparedStatement);
+        return JDBCUtil.getTableMetadata(preparedStatement, tablesAndViews);
+    }
+
+    @Override
+    protected ListTablesResponse listPaginatedTables(final Connection connection, final ListTablesRequest listTablesRequest) throws SQLException
+    {
+        String token = listTablesRequest.getNextToken();
+        int pageSize = listTablesRequest.getPageSize();
+
+        // no pagination case
+        if (pageSize == UNLIMITED_PAGE_SIZE_VALUE && token == null) {
```

Comment:
This segment of code id duplicated and this logic is common regardless of JDBC source type, can we move it 1 layer up as common logic in doListTables?